src/Apps/Collect/FilterState.tsx

```diff
@@ -169,11 +179,37 @@ export class FilterState extends Container<State> {
     return this.state.price_range !== "*-*"
   }
 
+  // generalize to avoid duplication?
```

Comment:
[Rule of threes](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) :)

Reply:
ha ok word. so I updated with generalized versions of these functions—they're functional but v inelegant. got any recs for learning to write more elegant JS/TS? open to revisions on this code ofc, but don't mean to make you spend time rewriting my code :)